### PR TITLE
ARROW-2935: [Packaging] Add verify_binary_artifacts function to verify-release-candidate.sh

### PR DESCRIPTION
**Pull request by cpcloud**

### CODE REVIEW

File: dev/release/02-source.sh [hunk 1]
```diff
@@ -97,18 +113,22 @@ ${SOURCE_DIR}/run-rat.sh ${tarball}
...
 svn co --depth=empty https://dist.apache.org/repos/dist/dev/arrow tmp
 
 # add the release candidate for the tag
-mkdir -p tmp/${tagrc}
+mkdir -p tmp/${tagrc}/binaries
+
+# copy the rc tarball into the tmp dir
 cp ${tarball}* tmp/${tagrc}
-svn add tmp/${tagrc}
-svn ci -m 'Apache Arrow ${version} RC${rc}' tmp/${tagrc}
```
Comment (cpcloud) [inline dev/release/02-source.sh]: Whoops, need to add these back.

File: dev/release/02-source.sh [hunk 2]
```diff
@@ -28,12 +28,28 @@ fi
...
 rc=$2
+artifact_dir=$3
 
 if [ -d tmp/ ]; then
   echo "Cannot run: tmp/ exists"
   exit
 fi
 
+if [ -z "$artifact_dir" ]; then
+    echo "artifact_dir is empty"
```
Comment (kou) [inline dev/release/02-source.sh]: Too much indent here.

File: dev/release/02-source.sh [hunk 3]
```diff
@@ -97,16 +113,22 @@ ${SOURCE_DIR}/run-rat.sh ${tarball}
...
 
 # add the release candidate for the tag
-mkdir -p tmp/${tagrc}
+mkdir -p tmp/${tagrc}/binaries
+
+# copy the rc tarball into the tmp dir
 cp ${tarball}* tmp/${tagrc}
+
+# copy binary artifacts into a subdirectory of the rc dir
+cp -rf "$artifact_dir"/* tmp/${tagrc}/binaries/*
```
Comment (kou) [inline dev/release/02-source.sh]: i think that `*` in `tmp/${tagrc}/binaries/*` is needless. It may cause a "glob not match" error.

**Review by kou — COMMENTED** [review]
I didn't run the change but I looks good to me except my comments.

File: dev/release/02-source.sh (hunk 3 above)
Comment (cpcloud) [reply dev/release/02-source.sh]: Ah good point. Thank you.

File: dev/release/02-source.sh (hunk 2 above)
Comment (cpcloud) [reply dev/release/02-source.sh]: Roger wilco.

**Review by kou — APPROVED** [review]
+1
I'll merge this.

**Pull request closed without merging**